### PR TITLE
Move `mysql` to BasicDatabaseClient

### LINKED ISSUE #1849: Move MySQL to the `BasicDatabaseClient` abstract class implementation
This will help us simplify everything.

We should have two final classes:
- MySqlClient
- MariaDbClient extends MySqlClient

There are basically 0 differences between MariaDB and MySQL, but this is a good convention we should follow.

### PR DESCRIPTION
**Pull request by azmy60**
closes #1849

### CODE REVIEW

**Comment by rathboma** [comment]
TODO: Check if removed functions are used anywhere

**Comment by rathboma** [comment]
@not-night-but once you've approved it tag @azmy60 so he can merge it!

File: apps/studio/src/lib/db/client.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ export interface IDbConnectionDatabase {
   connecting: boolean,
 }
 
-export class DBConnection {
+export class DBConnection implements DatabaseClient {
```
[the remark this answers is not in the document]
Comment (not-night-but) [reply apps/studio/src/lib/db/client.ts:159]: Yeah I think this is good!

File: apps/studio/src/lib/db/clients/mysql.ts [hunk 2]
```diff
@@ -0,0 +1,1340 @@
...
+    } else {
+      config.ssl.rejectUnauthorized = server.config.sslRejectUnauthorized;
+    }
+  }
+
+  return config;
+}
+
+function resolveDefault(
+  version: MysqlClient["versionInfo"],
```
Comment (not-night-but) [inline apps/studio/src/lib/db/clients/mysql.ts:167]: Did not know this was possible lol, but I think you should just have a class for versionInfo.

File: apps/studio/src/lib/db/clients/mysql.ts [hunk 3]
```diff
@@ -0,0 +1,1340 @@
...
+    defaultValue.length >= 2
+  ) {
+    // MariaDb escapes all single quotes with two single quotes in default value strings, even if they are
+    // escaped with backslashes in the original `CREATE TABLE` statement.
+    return defaultValue
+      .substring(1, defaultValue.length - 1)
+      .replaceAll("''", "'");
+  }
+
+  return defaultValue;
```
Comment (not-night-but) [inline apps/studio/src/lib/db/clients/mysql.ts:191]: We don't actually need to branch here. This function could just return `defaultValue` here, and then override it in the MariaDB class to do all that extra logic

File: apps/studio/src/lib/db/clients/mysql.ts [hunk 4]
```diff
@@ -0,0 +1,1340 @@
...
+}
+
+export class MysqlClient extends BasicDatabaseClient<ResultType> {
+  server: IDbConnectionServer;
+  database: IDbConnectionDatabase;
+  versionInfo: {
+    versionString: string;
+    version: number;
+    isMySql: boolean;
+    isMariaDb: boolean;
```
Comment (not-night-but) [inline apps/studio/src/lib/db/clients/mysql.ts:278]: If at all possible I think we should get rid of these. Any logic that isn't the same between the db's should be overridden in the child

File: apps/studio/src/lib/db/clients/mysql.ts [hunk 5]
```diff
@@ -0,0 +1,1340 @@
...
+    table: string,
+    orderBy: OrderBy[],
+    filters: string | TableFilter[],
+    chunkSize: number,
+    _schema?: string
+  ): Promise<StreamResults> {
+    const qs = buildSelectTopQuery(table, null, null, orderBy, filters);
+    const columns = await this.listTableColumns(db, table);
+    const rowCount = await this.getTableLength(table);
+    // TODO: DEBUG HERE
```
Comment (not-night-but) [inline apps/studio/src/lib/db/clients/mysql.ts:618]: Is this your todo?

**Review by not-night-but — CHANGES_REQUESTED** [review]
Looking good!! A couple things:
- Rather than branching on whether or not it is mysql, we should just override the function in the child
- Functions that need access to instance members, and don't need to be statically accessed can just be private/protected helper functions.

Fantastic job as always. Common Azmy W

File: apps/studio/src/lib/db/clients/mysql.ts (hunk 2 above)
Comment (azmy60) [reply apps/studio/src/lib/db/clients/mysql.ts:167]: Yeah, that makes total sense.

File: apps/studio/src/lib/db/clients/mysql.ts (hunk 5 above)
Comment (azmy60) [reply apps/studio/src/lib/db/clients/mysql.ts:618]: I don't think so.

**Review by wmontgomery — APPROVED** [review]

**Review by not-night-but — APPROVED** [review]
Looks good to me!

**Pull request merged**